src/framework/ui/internal/uiengine.cpp

```diff
@@ -67,6 +63,8 @@ void UiEngine::quit()
     m_engine->quit();
     delete m_engine;
     m_engine = nullptr;
+    delete m_theme;
+    m_theme = nullptr;
```

Comment:
this feels a bit more like a workaround than determining the root cause, as it wasn't happening with yesterday's code, but given it's crashing inside QtCore itself I'm not sure it's worth digging too much deeper

Reply:
There is already deleting of m_theme in UiEngine::~UiEngine(). I afraid this fix will hide some serious problems

Reply:
I left that in case a UIEngine is created and destroyed before quit() is called, but in normal circumstances it won't do anything.  Deleting m_theme in the destructor definitely causes a crash, I assume because something else in Qt has already been destroyed and made invalid.  But I don't know why it wasn't happening before yesterday.